[ADD] Issue templates for bug report and feature request

joao-p-marques opened the pull request:
As suggested in https://github.com/OCA/maintainer-quality-tools/issues/692

Usage example (similar): https://github.com/Tecnativa/doodba/issues/new/choose

simahawk reviewed (approved):
Excellent starting point. Thanks!

sbidoul reviewed (approved):
Shouldn't we explicitly ask about the OCA addon and Odoo series ?

Otherwise :+1:

joao-p-marques commented:
> Shouldn't we explicitly ask about the OCA addon and Odoo series ?

Thanks! Do you mean the Odoo and module version? We do ask for the affected versions [here](https://github.com/OCA/.github/pull/1/files#diff-185833cb26d7ac66a4d39042fd576a820c2c2c6d05ad18973bb9c7dce77267c5R16), although it could be more explicit, indeed...

We could also consider moving to [Issue Forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms) instead, although I am not sure these can be set at an organization level like this templates... 

What do you prefer?

yajo reviewed (approved):
We should merge now and fix issues later. This is an improvement already!

joao-p-marques commented:
Cool! Can we merge this, then? I will make the 2 PRs that @sbidoul asked in the [OCA Days chat](https://matrix.to/#/!gKuRXYxJkWkxMhIPXn:matrix.org/$_YviYr1JOXTVdoizyqEprGk3EQJexbG-qcCrEp1XoAU?via=recallstack.icu&via=t2bot.io&via=matrix.org) next week, but in the meantime this can be merged.

sbidoul commented on .github/ISSUE_TEMPLATE/config.yaml:
```diff
@@ -0,0 +1 @@
+blank_issues_enabled: false
```
what does this do ?

joao-p-marques replied:
This forces the user to choose one of the given options for the issue type and template (Bug Report or Feature Request). If set to `true`, the user would still be able to bypass that and open a "normal" issue: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser

sbidoul replied:
Should we not start with a softer approach ? And then disable it later when the issue template have proven to work well ?

joao-p-marques replied:
Ok, I changed it back to `true`.

joao-p-marques committed:
[ADD] Issue templates for bug report and feature request

As suggested in https://github.com/OCA/maintainer-quality-tools/issues/692

joao-p-marques commented:
@sbidoul I also added here a new PR template, where we can refer to the OCA contribution guidelines, instead of a per-project `CONTRIBUTING.md` file (removed in https://github.com/OCA/oca-addons-repo-template/pull/88)

sbidoul commented:
Great! I think [CONTRIBUTING.md is useful](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors) too, however.

sbidoul commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,27 @@
+assignees: ""
+---
+
+<!-- Provide a general summary of the issue in the Title above.
+Before opening the issue, please check that there isn't another issue addressing the same problem already created -->
+
```
```suggestion
<!-- Provide a general summary of the issue in the Title above.
Before opening the issue, please check that there isn't another issue addressing the same problem already created -->

## Module
The name of the module that has a bug.
```

sbidoul commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,7 @@
+assignees: ""
+---
+
+<!-- Please Please follow the official guide from the OCA Guidelines page (https://odoo-community.org/page/contributing)  -->
```
I suggest adding something link "please provide a summary of what your PR does and why it is necessary or useful."

joao-p-marques committed:
[ADD] CONTRIBUTING.md

Removed in https://github.com/OCA/oca-addons-repo-template/pull/88

joao-p-marques commented:
Oh ok. In that case, I added it here, and removed the PR template. I was trying to replace one with the other, but for OCA's use case it makes more sense to keep `CONTRIBUTING.md` only, indeed. The PR templates come handy when you want to be more specific when opening them but, as said, let's not introduce too much friction in the process at once.

sbidoul commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,10 @@
+Please follow the official guide from the
+[OCA Guidelines page](https://odoo-community.org/page/contributing).
+
+## Project Specific Guidelines
+
+<!-- /!\ do not modify above this line -->
+
+This project does not have specific coding guidelines.
```
Since this is now an org-level file, this section about project-specific guidelines does not make sense anymore.

joao-p-marques commented:
@sbidoul changes made. Thanks!

sbidoul commented:
LGTM, let's try this.

sbidoul commented:
@joao-p-marques do you want to communicate about this on the contributors mailing list ?

joao-p-marques commented:
> @joao-p-marques do you want to communicate about this on the contributors mailing list ?

Sure, I'll prepare the e-mail :+1:
